libsql-server/src/hrana/ws/session.rs

```diff
@@ -15,13 +15,26 @@ use crate::namespace::NamespaceName;
 
 /// Session-level state of an authenticated Hrana connection.
 pub struct Session {
-    auth: Authenticated,
-    version: Version,
+    pub auth: Authenticated,
```

Comment:
I think you could avoid leaking those details by having `handle_initial_hello`, `handle_repeated_hello` that both call private `handle_hello`.